src/AzureReadStream.cs

```diff
@@ -70,7 +73,23 @@ public override async Task<int> ReadAsync(byte[] buffer, int offset, int count,
             if (count > 0)
             {
                 await AzureRetry.Do(
-                    c => _blob.DownloadRangeToByteArrayAsync(buffer, offset, position, count, null, null, null, c),
+                    async c =>
+                    {
+                        var downloadUrl = _blob.GenerateSasUri(new BlobSasBuilder(BlobContainerSasPermissions.Read,
+                                                               new DateTimeOffset(DateTime.UtcNow.AddDays(1))));
+                        using (var httpClient = new HttpClient())
+                        {
+                            httpClient.DefaultRequestHeaders.Add("x-ms-range", $"bytes={position}-{offset + position + count - 1}");
```

Comment:
Unless I'm mistaken this should be `{position}-{position+count-1}` (this would ensure that we retrieve the `count` bytes we need, and `offset` is only related to `buffer`, not to the blob).